File: glean/lang/rust-lsif/tests/cases/xrefs/src/foo.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

// a module
// public type
#[derive(Debug)]
pub struct Foo {
    deep_foo: u64,
}

// public interface
pub fn new_foo(deep_foo: u64) -> Foo {
    new_secret_foo(deep_foo)
}

// not exported from the module
fn new_secret_foo(deep_foo: u64) -> Foo {
    Foo { deep_foo }
}

#[test]
fn main() {
    let foo = new_foo(42);
    println!("Foo: {:?}", foo);
}

File: glean/lang/codemarkup/tests/python/cases/search/some/foo.py
# Copyright (c) Meta Platforms, Inc. and affiliates.
# All rights reserved.
#
# This source code is licensed under the BSD-style license found in the
# LICENSE file in the root directory of this source tree.

# flake8: noqa
from .lib import a


class Foo:
    def meth(self, arg):
        a.some_func()


def foo():
    local_var = 1


foo_var = 1


class In1:
    class In2:
        class In3:
            class In4:
                class In5:
                    class In6:
                        class In7:
                            class In8:
                                class In9:
                                    def meth_9(self):
                                        return none

                                def meth_8(self):
                                    return none

                            def meth_7(self):
                                return none

                        def meth_6(self):
                            return none

                    def meth_5(self):
                        return none

                def meth_4(self):
                    return none

            def meth_3(self):
                return none

        def meth_2(self):
            return none

    def meth_1(self):
        return none
